File: android/practicaAndroid/PracticaI/app/src/main/java/holamundo/tutorial/com/practicai/MainActivity.java
package holamundo.tutorial.com.practicai;

import androidx.appcompat.app.AppCompatActivity;

import android.content.Intent;
import android.os.Bundle;
import android.os.Parcelable;
import android.util.Log;
import android.view.View;
import android.widget.TextView;
import android.widget.Toast;

import holamundo.tutorial.com.practicai.modelo.Persona;

public class MainActivity extends AppCompatActivity implements View.OnClickListener {

    private int i;

    TextView texto;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);

        texto= (TextView) findViewById(R.id.textoEvento);
     /*   texto.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View v) {
                Toast.makeText(MainActivity.this, "HAs hecho click con escuchador!..", Toast.LENGTH_SHORT).show();
            }
        });*/
     texto.setOnClickListener(this);

        Log.i("ciclo vida","ciclo vida: onCreate");
    }

    @Override
    protected void onStart() {
        super.onStart();
        Log.i("ciclo vida","ciclo vida: onStart");
    }

    @Override
    protected void onResume() {
        super.onResume();
        Log.i("ciclo vida","ciclo vida: onResume");
    }

    @Override
    protected void onPause() {
        super.onPause();
        Log.i("ciclo vida","ciclo vida: onPause");
    }


    public void initSecundarioActivity(View view){

        Toast.makeText(this, "Has hecho click en el texto", Toast.LENGTH_SHORT).show();
        
    }


    public void iniciarIntentActivity(View view){
        Persona p= new Persona();
        p.setNombre("Edwin");
        p.setApellido("Zamora");
        p.setDireccion("Ciudadela Ejercito");
        Bundle bundle= new Bundle();
        bundle.putSerializable("persona", p);


        Intent intent= new Intent(this,IntentActivity.class);
        intent.putExtra("numero",4);
        intent.putExtras(bundle);


        startActivity(intent);

    }


    @Override
    public void onClick(View v) {
        int id= v.getId();
        if (id== R.id.textoEvento) {
            Toast.makeText(this, "Has hecho click en el texto", Toast.LENGTH_SHORT).show();
        }
    }
}

File: kotlin/IntroduccionKotlin/app/src/main/java/dev/ezamora/cursokotlin/MainActivity.kt
package dev.ezamora.cursokotlin

import android.content.Intent
import android.os.Bundle
import android.view.View
import android.widget.Button
import android.widget.Toast
import androidx.appcompat.app.AppCompatActivity
import dev.ezamora.cursokotlin.google.calculator.CalculatorActivity
import dev.ezamora.cursokotlin.google.dice.DiceActivity
import dev.ezamora.cursokotlin.openwebinar.course01.example.list.ListOpenWebinarActivity
import dev.ezamora.cursokotlin.openwebinar.course01.basic.BasicConcept
import dev.ezamora.cursokotlin.openwebinar.course01.basic.Colecciones
import dev.ezamora.cursokotlin.openwebinar.course01.basic.Extenciones

import dev.ezamora.cursokotlin.openwebinar.course01.basic.ClassAndConstructor
import dev.ezamora.cursokotlin.openwebinar.course01.example.fuctionextencion.FuctionExtencionActivity
import dev.ezamora.cursokotlin.openwebinar.course01.lambdas.LambdasActivity
import dev.ezamora.cursokotlin.openwebinar.course01.recyclerView.ReciclerViewActivity
import kotlinx.android.synthetic.main.activity_main.*


class MainActivity : AppCompatActivity(), View.OnClickListener {
    val TAG = ":::TAG"
    var classAndConstructor = ClassAndConstructor();
    var basicConcept = BasicConcept();
    var extenciones = Extenciones();
    var colleciones = Colecciones();


    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)


        val btnListOpenWebinar: Button = findViewById(R.id.btnListOpenWebinar)
        val btnDice: Button = findViewById(R.id.btnDice)
        val btnCalculator: Button = findViewById(R.id.btnCalculator)
        val btnFntExtension: Button = findViewById(R.id.btnFntExtension)
        val btnLambdas: Button = findViewById(R.id.btnLambdas)
        val btnRecyclerView : Button= findViewById(R.id.btnRecyclerView)

        btnListOpenWebinar.setOnClickListener(this)
        btnDice.setOnClickListener(this)
        btnCalculator.setOnClickListener(this)
        btnFntExtension.setOnClickListener(this)
        btnLambdas.setOnClickListener(this)
        btnRecyclerView.setOnClickListener(this)

    }


    private fun basicConcept() {
        //  classAndConstructor.main()
        //  basicConcept.main()
        //extenciones.main()
        // colleciones.main()

    }

    override fun onClick(v: View?) {
        if (v != null) {
            if (v.id == btnListOpenWebinar.id) {
                //Toast.makeText(this, "HOliiiii", Toast.LENGTH_SHORT).show()
                val intent = Intent(this, ListOpenWebinarActivity::class.java).apply {
                }
                startActivity(intent)
            }
            if (v.id == btnDice.id) {
                val intent = Intent(this, DiceActivity::class.java).apply { }
                startActivity(intent)
            }
            if (v.id == btnCalculator.id) {
                val intent = Intent(this, CalculatorActivity::class.java).apply {}
                startActivity(intent)
            }
            if (v.id == btnFntExtension.id) {
                val intent = Intent(this, FuctionExtencionActivity::class.java).apply {}
                startActivity(intent)
            }
            if (v.id == btnLambdas.id) {
                val intent = Intent(this, LambdasActivity::class.java).apply { }
                startActivity(intent)
            }
            if (v.id == btnRecyclerView.id) {
              // Toast.makeText(this,"xxx",Toast.LENGTH_SHORT).show()
                val intent = Intent(this, ReciclerViewActivity::class.java).apply { }
                startActivity(intent)
            }
        }

    }


}


